$NetBSD: patch-CMakeLists.txt,v 1.5 2024/12/04 22:13:00 schmonz Exp $

HOMEBREW_HOME is provided by pkgsrc.
Split configuration between mysql-client and mysql-server.
Do not install documentation.

--- CMakeLists.txt.orig	2024-07-12 19:15:25.000000000 +0000
+++ CMakeLists.txt
@@ -1711,14 +1711,6 @@ SET(SYSTEM_LIBRARIES
   FIDO      # Homebrew  libfido2
   )
 
-IF(APPLE)
-  # Homebrew default location is
-  IF(APPLE_ARM)
-    SET(HOMEBREW_HOME "/opt/homebrew/opt")
-  ELSE()
-    SET(HOMEBREW_HOME "/usr/local/opt")
-  ENDIF()
-ENDIF()
 
 SET(WITH_SYSTEM_LIBS_DEFAULT OFF)
 OPTION(WITH_SYSTEM_LIBS
@@ -1956,7 +1948,7 @@ MYSQL_CHECK_RAPIDJSON()
 MYSQL_CHECK_FIDO()
 MYSQL_CHECK_FIDO_DLLS()
 
-IF(APPLE)
+IF(APPLE AND HOMEBREW_HOME)
   GET_FILENAME_COMPONENT(HOMEBREW_BASE ${HOMEBREW_HOME} DIRECTORY)
   IF(EXISTS ${HOMEBREW_BASE}/include/boost)
     FOREACH(SYSTEM_LIB ICU LIBEVENT LZ4 PROTOBUF ZSTD FIDO)
@@ -2232,11 +2224,12 @@ ENDIF()
 
 ADD_SUBDIRECTORY(client)
 ADD_SUBDIRECTORY(utilities)
-ADD_SUBDIRECTORY(share)
 ADD_SUBDIRECTORY(libservices)
 
-IF(NOT WITHOUT_SERVER)
+IF(WITHOUT_SERVER)
+  ADD_SUBDIRECTORY(share)
   ADD_SUBDIRECTORY(testclients)
+ELSE()
   ADD_SUBDIRECTORY(sql)
 ENDIF()
 
@@ -2336,7 +2329,7 @@ IF(ENABLE_GCOV)
   INCLUDE(fastcov)
 ENDIF()
 
-IF(UNIX)
+IF(UNIX AND WITHOUT_SERVER)
   ADD_SUBDIRECTORY(man)
 ENDIF()
 
@@ -2370,6 +2363,7 @@ SET_PROPERTY(DIRECTORY APPEND PROPERTY A
   "${CMAKE_BINARY_DIR}/library_output_directory"
   )
 
+IF(HAVE_BUILD_ID_SUPPORT AND NOT WITHOUT_SERVER)
 # Handle the "INFO_*" files.
 INCLUDE(${CMAKE_BINARY_DIR}/info_macros.cmake)
 # Source: This can be done during the cmake phase, all information is
@@ -2383,7 +2377,6 @@ MY_ADD_CUSTOM_TARGET(INFO_SRC ALL
   COMMENT "Generating Docs/INFO_SRC"
 )
 
-IF(HAVE_BUILD_ID_SUPPORT AND NOT WITHOUT_SERVER)
   # Show it by default in pushuild, but not for developers.
   IF(DEFINED ENV{PB2WORKDIR})
     OPTION(SHOW_BUILD_ID "Show mysqld build id as final target" ON)
@@ -2494,7 +2487,7 @@ ENDIF()
 #
 # RPM installs documentation directly from the source tree
 #
-IF(NOT INSTALL_LAYOUT MATCHES "RPM")
+IF(FALSE)
   INSTALL(FILES
     README
     LICENSE
